pkg/ui/workspaces/cluster-ui/src/databasesV2/index.tsx

```diff
@@ -200,6 +202,14 @@ export const DatabasesPageV2 = () => {
     sid => parseInt(sid, 10) as StoreID,
   );
 
+  // 409 conflict - this error code arises when the CRDB version
+  // is not compatible with the APIs used by this page.
+  // v2 Databases page is only supported for finalized CRDB versions >= 24.3.
+  // We can remove this check when the cluster-ui version moves past the 24.3 release.
+  if (!isCloud && error?.status === 409) {
```

Comment:
> pre-finalization, cloud marks the cluster as having the older version which would load the older UI code anyway

The cluster version in cloud was based off the post-finalized version last I checked, which is why we were also able to load the new pages prematurely, since the cluster-ui pkg loading code uses the same version var. 

To get around that we load the old pages based on the cluster version and whether or not the we've finalized the upgrade, which looks at a separate var. I threw in this boolean to be more clear on the redirect behaviour, since /legacy/databases doesn't exist in cloud, but yeah on cloud we should never be in the state of loading this code and having 409. If you prefer strongly to remove it I can do that.